lm_eval/caching/cache.py

```diff
@@ -0,0 +1,61 @@
+import os
+import dill
+import hashlib
+import random
+
+from lm_eval.utils import eval_logger
+
+MODULE_DIR = os.path.dirname(os.path.realpath(__file__))
+
+OVERRIDE_PATH = os.getenv("LM_HARNESS_CACHE_PATH")
+
+
+PATH = OVERRIDE_PATH if OVERRIDE_PATH else f"{MODULE_DIR}/.cache"
+
+# this is to effectively eliminate the possibility of a cache collision
+random.seed("EleutherAI")
```

Comment:
ah, actually--we don't want to be modifying the random seed for `random` here, as we need to respect #1412 .

Reply:
I just went ahead and removed the rand() seed, it was overkill anyway.

LMK if this is ready for a merge! Got all the tests and lint checker to pass on github :)

Reply:
Sorry for letting this sit, really appreciate the contribution! Merging now!